int64 encoderLeftCount
int64 encoderRightCount
float64 encoderLeftRate
float64 encoderRightRate
